### DISCUSSION TITLE
How does homebrew determine if a cask is installed?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#2045] [Everyday usage] [answered]
If I install cask `my-cask` with homebrew and then uninstall it via other means than homebrew, will homebrew still consider this cask installed?

What if it's a formula? I.e. I installed `my-formula` and then run `rm -rf /usr/local/opt/my-formula`, will homebrew still consider this formula installed?

**Comment by SMillerDev** [comment, marked as answer]
If you don't use brew, brew won't know you removed something.

**Reply by tddschn** [reply]
Thanks for the reply. I'm aware of this behavior, and I wonder 
- Does brew store which formulas / casks are installed somewhere, or does it compute the list when a related command is invoked?
- How do I manually mark a formula / cask as installed / not installed with brew?

**Reply by SMillerDev** [reply]
> Does brew store which formulas / casks are installed somewhere, or does it compute the list when a related command is invoked?

It stores files in various locations in the filesystem and some of those are checked when running things like `brew ls`

> How do I manually mark a formula / cask as installed / not installed with brew?

I guess you could see if manually removing any file related to a formula/cask works. But by far the easiest way is `brew install` and `brew rm`
